### ISSUE TITLE
Elementor won't Load when using Cloudflare' Rocket Loader

### ISSUE THREAD

**Issue by ianmubangizi** [pojome/elementor#2101]
I have been using CloudFlare for me SSL security, caching and Performance improvements. But not too long ago I noticed that my clients'' websites could not load the Elementor Page builder - but it seemed to only work in the Safari web browser - Chrome, Firefox, Internet Explore and The Edge browsers all seemed not to work even trying to update Java and enabling Flash and JavaScript. Well, observing the Log I noticed Chrome was blocking cloudflare.min.js as a violation. So this happened to be a problem caused by the Rocket Loader optimizer and disabling it fixed the problem and Elementor was working again.

**Comment by bainternet** [comment]
Basically, the issue is caused by rocket loader automatically changing the scripts to load asynchronously, which breaks the editor.

There are a few ways to get around it:

- Disable rocket loader 🤪

- Create Cloudflare Page rules to disable Rocket Loader only on specific URLS or conditions like `action=elementor` => a.k.a the editor and any other page it breaks the layout/functionality.

- Use `data-cfasync="false"` to tell Rocket Loader to skip a specific script from loading asynchronously.

This snippet adds `data-cfasync="false"` to all scripts in when loading the editor:

```PHP
<?php
add_filter( 'script_loader_tag', function( $tag, $handle ) {
	if ( isset( $_REQUEST['action'] ) && 'elementor' === $_REQUEST['action'] ) {
		return str_replace( ' src', ' data-cfasync="false" src', $tag );
	}
	return $tag;
}, 10, 2 );
```
@obe1968
Please let me kno if that helps, we'll considure adding this as a built-in solution.

**Comment by bainternet** [comment]
Closing for lack of activity,

Feel free to reopen if the issue persists.

**Comment by jaitheblogger** [comment]
Hello there.Where do i need to insert this snippet code in cloudflare??Inside cloudflare page rules i dont see any place to add this code.Kindly let me know.

**Comment by sheik124** [comment]
> 
> 
> Basically, the issue is caused by rocket loader automatically changing the scripts to load asynchronously, which breaks the editor.
> 
> There are a few ways to get around it:
> 
>     * Disable rocket loader 🤪
> 
>     * Create Cloudflare Page rules to disable Rocket Loader only on specific URLS or conditions like `action=elementor` => a.k.a the editor and any other page it breaks the layout/functionality.
> 
>     * Use `data-cfasync="false"` to tell Rocket Loader to skip a specific script from loading asynchronously.
> 
> 
> This snippet adds `data-cfasync="false"` to all scripts in when loading the editor:
> 
> ```
> <?php
> add_filter( 'script_loader_tag', function( $tag, $handle ) {
> 	if ( isset( $_REQUEST['action'] ) && 'elementor' === $_REQUEST['action'] ) {
> 		return str_replace( ' src', ' data-cfasync="false" src', $tag );
> 	}
> 	return $tag;
> }, 10, 2 );
> ```
> 
> @obe1968
> Please let me kno if that helps, we'll considure adding this as a built-in solution.

The PHP snippet does not work, nor does adding a very broad rule (*domain.org/wp-admin/*) to disable Rocket Loader.

Source map error: request failed with status 404
Resource URL: https://domain.org/wp-content/themes/charitious/assets/js/bootstrap.min.js?ver=1.0
Source Map URL: bootstrap.min.js.map[Learn More]
Source map error: request failed with status 404
Resource URL: https://domain.org/wp-content/themes/charitious/assets/js/Popper.js?ver=1.0
Source Map URL: popper.min.js.map[Learn More]

​

**Comment by mikeciffone** [comment]
It's been a while since this issue was closed, this has been published on Elementor's help center

For Elementor pages: `example.com/*elementor*`
For Elementor templates: `example.com/?elementor_library=*`

This seems to solve the editor issue. 

Testing last night on `pro 3.23.3` & `free 3.23.4` Rocket Loader almost works perfectly except menu drop-downs don't expand (using the `menu` widget, not `wordpress-menu`), and `off-canvas` elements don't open. I haven't tested it yet but I assume pop-up elements would be affected too.

I tried testing numerous page rules to exclude certain scripts for about an hour last night but had no luck. I'm guessing there's a specific dependency chain that needs to be left alone. Can someone enlighten me?
